### PR TITLE
fix: MessageDialog not working

### PR DESCRIPTION
**Pull request by nickrandolph**
GitHub Issue (If applicable): closes https://github.com/unoplatform/uno.extensions/issues/2275
fixes: #2184

## PR Type

What kind of change does this PR introduce?
- Bugfix

## What is the current behavior?

MessageDialog doesn't show on Uno 5.2

## What is the new behavior?

MessageDialog shows on Uno 5.2

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tested code with current [supported SDKs](../README.md#supported)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md). (for bug fixes / features) 
- [ ] [Unit Tests and/or UI Tests](doc/articles/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] [Wasm UI Tests](doc/articles/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences. Validate PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Updated the [Release Notes](https://github.com/unoplatform/uno/tree/master/doc/ReleaseNotes)
- [ ] Associated with an issue (GitHub or internal)

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Comment by nickrandolph** [comment]
@mergify backport release/stable/4.1

File: samples/Playground/Playground/Playground.csproj [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <TargetFrameworks Condition="$([MSBuild]::IsOSPlatform('windows')) or '$(EnableWindowsTargeting)' == 'true'">$(TargetFrameworks);net8.0-windows10.0.19041</TargetFrameworks>
-    <TargetFrameworks>$(TargetFrameworks);net8.0;net8.0-android;net8.0-ios;net8.0-maccatalyst</TargetFrameworks>
+    <!--<TargetFrameworks>$(TargetFrameworks);net8.0;net8.0-android;net8.0-ios;net8.0-maccatalyst</TargetFrameworks>-->
```
Comment (agneszitte) [inline samples/Playground/Playground/Playground.csproj:4]: Are we leaving this commented here?

Comment (nickrandolph) [reply samples/Playground/Playground/Playground.csproj:4]: oh, good catch - I'll revert this change

Comment (nickrandolph) [inline samples/Playground/Playground/Playground.csproj:4]:
```suggestion
    <TargetFrameworks>$(TargetFrameworks);net8.0;net8.0-android;net8.0-ios;net8.0-maccatalyst</TargetFrameworks>
```

**Review by dansiegel — APPROVED** [review]

File: samples/Playground/Playground/Playground.csproj (hunk 1 above)
Comment (jeromelaban) [inline samples/Playground/Playground/Playground.csproj:4]: @nickrandolph is this still required?

Comment (nickrandolph) [reply samples/Playground/Playground/Playground.csproj]: uncommented

File: src/Uno.Extensions.Navigation.UI/Navigators/MessageDialogNavigator.cs [hunk 2]
```diff
@@ -95,7 +95,7 @@ protected override Task<bool> RegionCanNavigate(Route route, RouteInfo? routeMap
 		};
 		md.Commands.AddRange(commands);
 
-#if WINUI && WINDOWS
+#if WINUI
 		var window = Window!;
 		var hwnd = WinRT.Interop.WindowNative.GetWindowHandle(window);
 		WinRT.Interop.InitializeWithWindow.Initialize(md, hwnd);
```
Comment (MartinZikmund) [inline src/Uno.Extensions.Navigation.UI/Navigators/MessageDialogNavigator.cs:101]: The condition here should likely be `#if WINUI || HAS_UNO_WINUI`. On targets with multiwindow support, `MessageDialog` also now requires this `InitializeWithWindow` snippet

**Review by MartinZikmund — APPROVED** [review]

File: testing/TestHarness/TestHarness.Windows/TestHarness.Windows.csproj [hunk 3]
```diff
@@ -53,6 +53,7 @@
 		<PackageReference Include="Uno.Toolkit.WinUI.Material" />
 		<PackageReference Include="Uno.Toolkit.WinUI" />
 		<PackageReference Include="Uno.WinUI.MSAL" />
+		<PackageReference Include="Uno.WinUI" />
```
Comment (carldebilly) [inline testing/TestHarness/TestHarness.Windows/TestHarness.Windows.csproj:56]: I'm curious why this reference is needed here...

**Review by carldebilly — APPROVED** [review]

File: testing/TestHarness/TestHarness.Windows/TestHarness.Windows.csproj (hunk 3 above)
Comment (nickrandolph) [reply testing/TestHarness/TestHarness.Windows/TestHarness.Windows.csproj:56]: This was an attempt to try to fix the build error

File: src/Uno.Extensions.Navigation.UI/Navigators/MessageDialogNavigator.cs (hunk 2 above)
Comment (nickrandolph) [reply src/Uno.Extensions.Navigation.UI/Navigators/MessageDialogNavigator.cs]: Updated

**Pull request merged**